src/number_theory/modular_forms/slash_actions.lean

```diff
@@ -26,38 +25,38 @@ local notation `GL(` n `, ` R `)`⁺ := matrix.GL_pos (fin n) R
 local notation `SL(` n `, ` R `)` := matrix.special_linear_group (fin n) R
 
 /--A general version of the slash action of the space of modular forms.-/
-class slash_action (β : Type*) (G : Type*) (α : Type*) (γ : Type*) [group G] [ring α]
-  [has_smul γ α] :=
-(map : β → G → α → α)
+class slash_action (β G α γ : Type*) [group G] [has_zero α ]
+[has_one α ] [has_smul γ α ] [has_add α ] :=
+(map : β → G → α → α )
```

Comment:
```suggestion
class slash_action (β G α γ : Type*) [group G] [has_zero α]
[has_one α] [has_smul γ α] [has_add α] :=
(map : β → G → α → α)
```
Same below on `monoid_hom_slash_action`; did something happen to add these spaces back in?